src/test/java/com/laserfiche/api/client/unit/UsernamePasswordHandlerTest.java

```diff
@@ -0,0 +1,103 @@
+
+public class UsernamePasswordHandlerTest extends BaseTest {
+    private Request _request = new RequestImpl();
+
+    @Test
+    @Disabled("Throwing a null exception -> will have to figure out how to stub completable future type functions")
+    void beforeSendAsync_NewToken_Success() {
+        // Arrange
+        String accessToken = "access_token";
+        SessionKeyInfo mockedResponse = new SessionKeyInfo();
+        mockedResponse.setAccessToken(accessToken);
+        CreateConnectionRequest mockedBody = mock(CreateConnectionRequest.class);
+        TokenClient mockedClient = mock(TokenClientImpl.class);
+        when(mockedClient.createAccessToken(eq(anyString()), mockedBody)).thenReturn(
+                CompletableFuture.completedFuture(mockedResponse));
+        HttpRequestHandler handler = new UsernamePasswordHandler(repoId, username, password, baseUrl, mockedClient);
+
+        //Act
+        BeforeSendResult result = handler
+                .beforeSendAsync(_request)
+                .join();
+
+        //Assert
+        assertNotNull(result);
+        assertNull(result.getRegionalDomain());
+        assertTrue(_request
+                .headers()
+                .get("Authorization")
+                .contains("Bearer"));
+        assertNotNull(_request
+                .headers()
+                .get("Authorization")
+                .substring(6, _request
+                        .headers()
+                        .get("Authorization")
+                        .length() - 1));
+        verify(mockedClient, times(1)).createAccessToken(anyString(), mockedBody);
+    }
+
+    @Test
+    void afterSendAsync_ResponseUnauthorized_ReturnsTrue() {
+        // Arrange
+        HttpRequestHandler handler = new UsernamePasswordHandler(repoId, username, password, baseUrl, null);
+        Response mockedResponse = mock(Response.class);
+        when(mockedResponse.status()).thenReturn((short) 401);
+
+        // Act
+        handler
+                .afterSendAsync(mockedResponse)
+                .thenApply((shouldRetry) -> {
+                    // Assert
+                    assertEquals(true, shouldRetry);
+                    return null;
+                });
+    }
+
+    @ParameterizedTest
+    @MethodSource("ResponseOtherThanUnauthorized")
+    void afterSendAsync_ResponseOtherThanUnauthorized_ReturnsFalse(int status) {
+        // Arrange
+        HttpRequestHandler handler = new UsernamePasswordHandler(repoId, username, password, baseUrl, null);
+        Response mockedResponse = mock(Response.class);
+        when(mockedResponse.status()).thenReturn((short) status);
+
+        // Act
+        handler
+                .afterSendAsync(mockedResponse)
+                .thenApply((shouldRetry) -> {
+                    // Assert
+                    assertEquals(false, shouldRetry);
+                    return null;
+                });
+    }
+
+    private static Stream<Arguments> ResponseOtherThanUnauthorized() {
```

Comment:
In Java, method name starts with lowercase letter.

Reply:
done